source/locationHelper.py

```diff
@@ -0,0 +1,398 @@
+
+	def __gt__(self,other):
+		"""Returns whether this rectangle is a superset of other (i.e. whether all points of other are contained by this rectangle)."""
+		if not isinstance(other,RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return self>=other and self!=other
+
+	def __ge__(self,other):
+		"""Returns whether self is a superset of or equal to other."""
+		if not isinstance(other,RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return other.left >= self.left and other.top >= self.top and other.right <= self.right and other.bottom <= self.bottom
+
+	def __eq__(self,other):
+		if not isinstance(other,RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return other.left == self.left and other.top == self.top and other.right == self.right and other.bottom == self.bottom
+
+	def __neq__(self,other):
+		return not (self==other)
+
+	def __sub__(self,other):
+		if not isinstance(other,RECT_CLASSES):
+			return NotImplemented
+		left,top,right,bottom=self.left-other.left,self.top-other.top,self.right-other.right,self.bottom-other.bottom
+		if isinstance(self, Location):
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def __rsub__(self,other):
+		return self.__sub__(other)
+
+	def __and__(self,other):
+		"""Returns the intersection of self and other.
+		For example, if self = Rect(left=10,top=10,right=25,bottom=25) and other = Rect(left=20,top=20,right=35,bottom=35),
+		this results in Rect(left=20,top=20,right=25,bottom=25).
+		No intersect results in a rectangle with zeroed coordinates.
+		"""
+		if not isinstance(other,RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return NotImplemented
+		left=max(self.left,other.left)
+		top=max(self.top,other.top)
+		right=min(self.right,other.right)
+		bottom=min(self.bottom,other.bottom)
+		if left>right or top>bottom:
+			left=top=right=bottom=0
+		if isinstance(self, Location):
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def __rand__(self,other):
+		return self.__and__(other)
+
+class Location(_RectMixin, namedtuple("Location",("left","top","width","height"))):
+	"""
+	Represents a rectangle on the screen, based on left and top coordinates, width and height.
+	To represent a rectangle using left, top, right and bottom coordinates, use L{Rect}.
+	"""
+
+	@property
+	def right(self):
+		return self.left+self.width
+
+	@property
+	def bottom(self):
+		return self.top+self.height
+
+	def toRect(self):
+		return Rect(self.left,self.top,self.right,self.bottom)
+
+class Rect(_RectMixin, namedtuple("Rect",("left","top","right","bottom"))):
+	"""Represents a rectangle on the screen.
+	By convention, the right and bottom edges of the rectangle are normally considered exclusive.

```

Comment:
Could you clarify what this means with an example? Or better, point to a unit test that demonstrates this.

Reply:
This is a convention followed by NVDA, as this convension is also followed by Windows. I will document this with references. I'm afraid unit tests do not make sense here.

Reply:
I would remove the words "normally considered", unless there is some case where they are not exclusive? A unit test could show that given `Rect(0, 0, 1, 1)` then `Point(0,0)` is inside, but `Point(1,1)`, `Point(0,1)`, `Point(1,0)` aren't.